Python/optimizer_bytecodes.c

```diff
@@ -1216,6 +1216,18 @@ dummy_func(void) {
         sym_set_const(callable, list_append);
     }
 
+    op(_BINARY_SLICE, (container, start, stop -- res)) {
+        // Slicing a string/list/tuple always returns the same type.
+        PyTypeObject *type = sym_get_type(container);
+        if (type == &PyUnicode_Type ||
+            type == &PyList_Type ||
+            type == &PyTuple_Type) {
+            res = sym_new_type(ctx, type);
+        } else {
```

Comment:
I know CPython's brace style is a bit weird to get used to, but we should follow PEP 7:

```suggestion
            type == &PyTuple_Type)
        {
            res = sym_new_type(ctx, type);
        }
        else {
```